File: programacion/java/2_unidad/1/Ejercicio2.java
import java.util.Scanner;
// sacar todos los números primos entre dos números introducidos por teclado (ambos inclusive)
public class Ejercicio2 {
    public static void main(String[] args) {
        Scanner keyboard = new Scanner(System.in);

        System.out.print("Introduce el primer número: ");
        int num1 = Integer.parseInt(keyboard.nextLine());
        System.out.print("Introduce el segundo número: ");
        int num2 = Integer.parseInt(keyboard.nextLine());
        boolean esPrimo;

        if (num1 < 1 || num2 < 1 || num1 == num2) {
            System.out.println("Ambos números deben ser mayor de 0 y distintos.");
        } else {
            int menor = Math.min(num1, num2);
            int mayor = Math.max(num1, num2);
            System.out.println("Números primos entre " + menor + " y " + mayor + ":");

            for (int i = menor; i <= mayor; i++) {
                esPrimo = true;

                for (int j = 2; j <= Math.sqrt(i); j++) {
                    
                    if (i % j == 0) {
                        esPrimo = false;
                        break;
                    }
                }
                
                if (esPrimo && i > 1) {
                    System.out.println(i);
                }
            }
        }
        System.out.println("\nFin del programa.");
    }
}

File: programacion/python/unidad_2/Ejercicio2.py
import math
num1 = int(input("Ingrese el primer número: "))
num2 = int(input("Ingrese el segundo número: "))
esPrimo = True

if num1 < 1 or num2 < 1 or num1 == num2:
    print("Error: Los números deben ser positivos y diferentes.")
else:
    
    menor = min(num1, num2)
    mayor = max(num1, num2)
    print(f"Números primos entre {menor} y {mayor}:")
    
    for i in range(menor, mayor +1):
        esPrimo = True
        
        for j in range(2, int(math.sqrt(i)) + 1):
            
            if i % j == 0:
                esPrimo = False
                break
            
        if esPrimo and i > 1:
            print(i)
print("Fin del programa.")